### PR TITLE
add an option of using aqt dq dg

### PR DESCRIPTION
**Pull request by ychzhang**

### CODE REVIEW

File: MaxText/configs/base.yml [hunk 1]
```diff
@@ -40,6 +40,7 @@ gcs_metrics: False
 # Activation dtypes.
 dtype: "bfloat16"
 int8_training: False
+use_dqdg: False
```
Comment (lukaszlew) [inline MaxText/configs/base.yml:43]: Add a comment that it has effect only if int8_training=True on this and 2 flags belof

File: MaxText/layers.py [hunk 2]
```diff
@@ -225,12 +226,16 @@ def __call__(self, inputs: Array) -> Array:
     if not cfg.int8_training:
       return lax.dot_general(inputs, kernel, ((axis, contract_ind), ((), ())))
     else:
-      aqt_cfg = maxtext_sweeps.sweep1(cfg.fwd_int8, cfg.bwd_int8)
-      aqt_dot_general = aqt.make_dot_general(aqt_cfg)
       aqt_key = self.make_rng('aqt')
-      context = aqt.Context(key=aqt_key, train_step=None)
+      if cfg.use_dqdg:
+        aqt_dq_dg = aqt_dq.make_aqt_dq_dg()
+        return aqt_dq_dg(aqt_key, inputs, kernel, ((axis, contract_ind), ((), ())))
```
Comment (lukaszlew) [inline MaxText/layers.py:232]: how come it does not need rng?

**Review by lukaszlew — CHANGES_REQUESTED** [review]

File: MaxText/layers.py (hunk 2 above)
Comment (ychzhang) [reply MaxText/layers.py:232]: It needs. Rng is the first argument of this function (not in Context). Whereas rng for aqt v2 is wrapped in a context object.

Comment (lukaszlew) [reply MaxText/layers.py:232]: I'm blind :)

**Pull request merged**